sklearn/utils/__init__.py

```diff
@@ -178,22 +178,63 @@ def axis0_safe_slice(X, mask, len_mask):
     return np.zeros(shape=(0, X.shape[1]))
 
 
-def safe_indexing(X, indices):
+def safe_indexing(X, indices, axis=0):
```

Comment:
sorry I didn't look in details again since the last changes: do you also test that axis=1 raises an error in case X isn't 2d or X is a series?